src/lib/components/references/bibleTextReferenceFetcher.ts

```diff
@@ -20,191 +20,108 @@ export const fetchAndFormat = async (
     language: Language,
     passedBibleId?: number
 ): Promise<BibleTextsReference | null> => {
+    if (startVerse > endVerse) {
+        log.exception(new Error(`Unexpected verseIds: startVerse: ${startVerse}; endVerse: ${endVerse}`));
+        return null;
+    }
+
     // ensure we always include verse 0 along with verse 1 if verse 0 exists
     if (startVerse % 1000 === 1) {
         startVerse -= 1;
     }
 
+    // if no Bible was passed then get the default Bible for the language with a fallback of 1
     const bibleId =
         !passedBibleId || passedBibleId === 0
             ? ((await fetchLanguageDefaultBible(language.id))?.id ?? 1)
             : passedBibleId;
 
-    const [bookTexts, versificationMappings] = await Promise.all([
-        fetchBiblePassages(startVerse, endVerse, bibleId),
```

Comment:
Fetching `BookTexts` is now done _after_ fetching versification mappings so we don't needlessly fetch the wrong chapter (e.g. Psalm 11 in ENG is Psalm 10 in RSB) and then have to refetch the correct chapter.  This also simplified the flow.